crypto/src/utils.rs

```diff
@@ -0,0 +1,197 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(dead_code)] // TODO Remove after complete ECR-2518 and  ECR-2519
+use super::{gen_keypair, PublicKey, SecretKey};
+use pwbox::{sodium::Sodium, ErasedPwBox, Eraser, Suite};
+use rand::thread_rng;
+#[cfg(unix)]
+use std::os::unix::fs::{MetadataExt, OpenOptionsExt};
+use std::{
+    fs::{File, OpenOptions},
+    io::{Error, ErrorKind, Read, Write},
+    path::Path,
+};
+use toml;
+
+/// Creates a TOML file that contains encrypted `SecretKey` and returns `PublicKey` for the secret key.
+pub fn create_keys_file<P: AsRef<Path>>(path: P, pass_phrase: &[u8]) -> Result<PublicKey, Error> {
+    let (pk, sk) = gen_keypair();
+    let sk_bytes = hex::decode(sk.to_hex()).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    let keys = encrypt(pk.to_hex(), &sk_bytes, pass_phrase)?;
+    let file_content =
+        toml::to_string_pretty(&keys).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    let mut open_options = OpenOptions::new();
+    open_options.create(true).write(true);
+    #[cfg(unix)]
+    open_options.mode(0o600);
+    let mut file = open_options.open(path.as_ref())?;
+    file.write_all(file_content.as_bytes())?;
+
+    Ok(pk)
+}
+
+/// Reads `PublicKey` and `SecretKey` from encrypted file located by path and returns its.
+pub fn read_keys_from_file<P: AsRef<Path>>(
+    path: P,
+    pass_phrase: &[u8],
+) -> Result<(PublicKey, SecretKey), Error> {
+    let mut key_file = File::open(path)?;
+
+    if cfg!(unix) {
+        let file_info = key_file.metadata()?;
+        if (file_info.mode() & 0o600) != 0o600 {
+            return Err(Error::new(ErrorKind::Other, "Wrong file's mode"));
+        }
+    }
+
+    let mut file_content = vec![];
+    key_file.read_to_end(&mut file_content)?;
+    let keys: EncryptedKeys =
+        toml::from_slice(file_content.as_slice()).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    let pub_key = PublicKey::from_slice(
+        &hex::decode(&keys.public_key).map_err(|e| Error::new(ErrorKind::Other, e))?,
+    ).ok_or(Error::new(
+        ErrorKind::Other,
+        "Couldn't create PublicKey from slice",
+    ))?;
+    let sec_key = SecretKey::from_slice(&decrypt(&keys, pass_phrase)?).ok_or(Error::new(
+        ErrorKind::Other,
+        "Couldn't create SecretKey from slice",
+    ))?;
+
+    Ok((pub_key, sec_key))
+}
+
+#[derive(Serialize, Deserialize)]
+struct EncryptedKeys {
+    public_key: String,
+    secret_key: ErasedPwBox,
+}
+
+fn encrypt(
+    public_key: String,
+    secret_key: &[u8],
+    pass_phrase: &[u8],
+) -> Result<EncryptedKeys, Error> {
+    let mut rng = thread_rng();
```

Comment:
Nit / off-topic: I think it could be possible to implement a CSRNG based on `libsodium`'s [`randombytes`](https://docs.rs/exonum_sodiumoxide/0.0.20/exonum_sodiumoxide/randombytes/fn.randombytes_into.html) and maybe, even include it into the library. Not sure whether it makes much sense though.